### PR TITLE
Move Snapshot class from bootstrap to main module

### PR DESCRIPTION
**Pull request by jpbempel**
# What Does This Do
Move `CapturedContext` as Top level class remaining in bootstrap 
Remove `Sink` interface from `DebuggerContext` and use it directly from `DebuggerAgent`
Remove `ProbeDiagnostic`, not used
Move `ProbeLocation` and `EvaluationError` as top level class 
Rename `SnapshotSerializer` as `ValueSerializer` and remove `serializeSnapshot` method. Snapshot are serialized directly from `DebuggerAgent`'s `JsonSnapshotSerializer`

# Motivation
Shrink down the bootstrap project and cut some dependencies 
# Additional Notes

### CODE REVIEW

File: dd-java-agent/agent-debugger/src/main/java/com/datadog/debugger/agent/DebuggerAgent.java [hunk 1]
```diff
@@ -157,11 +166,25 @@ static void stop() {
     if (configurationPoller != null) {
       configurationPoller.stop();
     }
-    if (sink != null) {
-      sink.stop();
```
Comment (shatzi) [inline dd-java-agent/agent-debugger/src/main/java/com/datadog/debugger/agent/DebuggerAgent.java:161]: maybe add stop to the Sink interface?

**Review by shatzi — APPROVED** [review]
Great stuff, thank you!

File: dd-java-agent/agent-debugger/debugger-bootstrap/src/main/java/com/datadog/debugger/sink/Snapshot.java [hunk 2]
```diff
@@ -0,0 +1,296 @@
...
+      stack.add(CapturedStackFrame.from(ste));
+    }
+  }
+
+  public enum Kind {
+    ENTER,
+    RETURN,
+    UNHANDLED_EXCEPTION,
+    HANDLED_EXCEPTION,
+    BEFORE,
```
[the remark this answers is not in the document]
Comment (jpbempel) [reply dd-java-agent/agent-debugger/debugger-bootstrap/src/main/java/com/datadog/debugger/sink/Snapshot.java:175]: no I don't think so, legacy

File: dd-java-agent/agent-debugger/debugger-bootstrap/src/main/java/datadog/trace/bootstrap/debugger/DebuggerContext.java [hunk 3]
```diff
@@ -48,21 +40,17 @@ public interface Tracer {
     DebuggerSpan createSpan(String resourceName, String[] tags);
   }
 
-  public interface SnapshotSerializer {
-    String serializeSnapshot(String serviceName, Snapshot snapshot);
-
-    String serializeValue(Snapshot.CapturedValue value);
+  public interface ValueSerializer {
```
[the remark this answers is not in the document]
Comment (jpbempel) [reply dd-java-agent/agent-debugger/debugger-bootstrap/src/main/java/datadog/trace/bootstrap/debugger/DebuggerContext.java:43]: to serialize CapturedValue from the CapturedValue.freeze

File: dd-java-agent/agent-debugger/src/main/java/com/datadog/debugger/agent/DebuggerAgent.java [hunk 4]
```diff
@@ -49,19 +51,21 @@ public static synchronized void run(
     ddAgentFeaturesDiscovery.discoverIfOutdated();
     agentVersion = ddAgentFeaturesDiscovery.getVersion();
 
-    sink = new DebuggerSink(config);
-    sink.start();
+    DebuggerSink debuggerSink = new DebuggerSink(config);
```
[the remark this answers is not in the document]
Comment (jpbempel) [reply dd-java-agent/agent-debugger/src/main/java/com/datadog/debugger/agent/DebuggerAgent.java:54]: need to call som method specific to DebuggerSink

File: dd-java-agent/agent-debugger/src/main/java/com/datadog/debugger/agent/DebuggerAgent.java (hunk 1 above)
Comment (jpbempel) [reply dd-java-agent/agent-debugger/src/main/java/com/datadog/debugger/agent/DebuggerAgent.java:161]: sounds like implementation details, leaking abstraction...

File: dd-java-agent/agent-debugger/debugger-bootstrap/src/main/java/com/datadog/debugger/sink/Snapshot.java
[the remark this answers is not in the document]
Comment (jpbempel) [reply dd-java-agent/agent-debugger/debugger-bootstrap/src/main/java/com/datadog/debugger/sink/Snapshot.java:1]: done

**Pull request merged**